src/dealWizard/wizardManager.ts

```diff
@@ -0,0 +1,116 @@
+import { autoinject } from "aurelia-framework";
+import { PLATFORM } from "aurelia-pal";
+import { RouteConfig } from "aurelia-router";
+import { WizardService, IWizardState, IWizardStage } from "services/WizardService";
+import { DealRegistrationData, IDealRegistrationData } from "entities/DealRegistrationData";
+import { IStageMeta, WizardType, STAGE_ROUTE_PARAMETER } from "./dealWizardTypes";
+
+const proposalStage: IWizardStage = {
+  name: "Proposal",
+  valid: false,
+  route: "proposal",
+  moduleId: PLATFORM.moduleName("./stages/proposalStage/proposalStage"),
+};
+
+const proposalLeadStage: IWizardStage = {
+  name: "Lead Details",
+  valid: false,
+  route: "proposal-lead",
+  moduleId: PLATFORM.moduleName("./openProposalWizard/openProposalProposalLeadStage/openProposalProposalLeadStage"),
+};
+
+const primaryDaoStage: IWizardStage = {
+  name: "Primary DAO",
+  valid: false,
+  route: "primary-dao",
+  moduleId: PLATFORM.moduleName("./stages/primaryDaoStage/primaryDaoStage"),
+};
+
+const partnerDaoStage: IWizardStage = {
+  name: "Partner DAO",
+  valid: false,
+  route: "partner-dao",
+  moduleId: PLATFORM.moduleName("./stages/partnerDaoStage/partnerDaoStage"),
+};
+
+@autoinject
+export class WizardManager {
+  public wizardState: IWizardState<IDealRegistrationData>;
+  public stageMeta: IStageMeta;
+  public view: string;
+  public viewModel: string;
+  private stages: IWizardStage[] = [];
+  private registrationData = new DealRegistrationData();
+
+  constructor(public wizardService: WizardService) {}
+
+  activate(params: {[STAGE_ROUTE_PARAMETER]: string}, routeConfig: RouteConfig): void {
+    if (!params[STAGE_ROUTE_PARAMETER]) return;
+
+    const stageRoute = params[STAGE_ROUTE_PARAMETER];
+    const wizardType = routeConfig.settings.wizardType;
+    const parentRoutePath = routeConfig.route as string;
+
+    if (!this.wizardService.hasWizard(this)) {
```

Comment:
I think it would be better if the service makes this check inside the function

Reply:
Good catch, actually it's not necessary to have this if statement at all. I will remove it